R/find.tol.time.R

```diff
@@ -1,86 +1,102 @@
-        da<-0
-        if(length(breaks)>max.num.segments)
-        {
-            s<-floor(seq(2, length(breaks), length.out=max.num.segments))
-        }else{
-            s<-2:length(breaks)
+    breaks <- breaks[c(-1, -length(breaks))]
+    if (is.na(rt_tol_relative)) {
+        da <- 0
+        if (length(breaks) > max.num.segments) {
+            s <- floor(seq(2, length(breaks), length.out = max.num.segments))
+        } else {
+            s <- 2:length(breaks)
         }
         
-        for(i in s)
-        {
-            this.sel<-(breaks[i-1]+1):breaks[i]
+        for (i in s) {
+            this.sel <- (breaks[i - 1] + 1):breaks[i]
             
-            if(length(this.sel) <= 3*num.exp)
-            {
-                this.d<-as.vector(dist(chr[this.sel]))
-                if(length(this.d)>100) this.d<-sample(this.d,100)
-                da<-c(da,this.d)
+            if (length(this.sel) <= 3 * number_of_experiments) {
+                this.d <- as.vector(dist(chr[this.sel]))
+                if (length(this.d) > 100)
+                    this.d <- sample(this.d, 100)
+                da <- c(da, this.d)
             }
         }
         
-        da<-da[!is.na(da)]
-        uppermost<-max(da)
-        n=min(max.bins,max(min.bins, round(length(da)/aver.bin.size)))
-        des<-density(da,kernel="gaussian",n=n, bw=uppermost/n*2,from=0)
-        y<-des$y[des$x>0]
-        x<-des$x[des$x>0]
+        da <- da[!is.na(da)]
+        uppermost <- max(da)
+        n = min(max.bins, max(min.bins, round(length(da) / aver.bin.size)))
+        des <- density(da, kernel = "gaussian", n = n,
+                       bw = uppermost / n * 2, from = 0)
+        y <- des$y[des$x > 0]
+        x <- des$x[des$x > 0]
         
-        this.l<-lm(y[x>uppermost/4]~x[x>uppermost/4])
+        this.l <- lm(y[x > uppermost / 4] ~ x[x > uppermost / 4])
         
-        exp.y<-this.l$coef[1]+this.l$coef[2]*x
-
-        y2<-y[1:(length(y)-1)]
-        y3<-y[2:(length(y))]
-        y2[which(y2<y3)]<-y3[which(y2<y3)]
-        y[1:(length(y)-1)]<-y2
+        exp.y <- this.l$coef[1] + this.l$coef[2] * x
+        
+        y2 <- y[1:(length(y) - 1)]
+        y3 <- y[2:(length(y))]
+        y2[which(y2 < y3)] <- y3[which(y2 < y3)]
+        y[1:(length(y) - 1)] <- y2
+        
+        yy <- cumsum(y > 1.5 * exp.y)
+        yi <- seq_along(yy)
+        sel <- min(which(yy < yi)) - 1
+        rt_tol_relative <- x[sel]
         
-        yy<-cumsum(y>1.5*exp.y)
-        yi<- seq_along(yy)
-        sel<-min(which(yy<yi))-1
-        chr.tol<-x[sel]
-
         if (do.plot) {
-            plot(x,y,main="find retention time tolerance", xlab="Delta", ylab="Density",cex=0.25)
-            lines(x,exp.y,col="red")
-            abline(v=x[sel],col="blue")
+            plot(x, y,  xlab = "Delta", ylab = "Density",
+                 main = "find retention time tolerance", cex = 0.25)
+            lines(x, exp.y, col = "red")
+            abline(v = x[sel], col = "blue")
```

Comment:
plotting...

Reply:
Extracted in [ee5a295](https://github.com/RECETOX/recetox-aplcms/pull/63/commits/ee5a2950c9de8fad8d302a8321fc5c21cb0ca213).